src/compiler/utilities.ts

```diff
@@ -4179,14 +4182,14 @@ export function canHaveJSDoc(node: Node): node is HasJSDoc {
 export function getJSDocCommentsAndTags(hostNode: Node, noCache?: boolean): readonly (JSDoc | JSDocTag)[] {
     let result: (JSDoc | JSDocTag)[] | undefined;
     // Pull parameter comments from declaring function as well
-    if (isVariableLike(hostNode) && hasInitializer(hostNode) && hasJSDocNodes(hostNode.initializer!)) {
-        result = addRange(result, filterOwnedJSDocTags(hostNode, last((hostNode.initializer as HasJSDoc).jsDoc!)));
+    if (isVariableLike(hostNode) && hasInitializer(hostNode) && hostNode.initializer && hasJSDocNodes(hostNode.initializer) && hostNode.initializer.jsDoc) {
+        result = addRange(result, filterOwnedJSDocTags(hostNode, hostNode.initializer.jsDoc));

```

Comment:
I don't want to change the behaviour of other tags, which are only valid in the jsdoc immediately before a node.

When I changed this before (4 years ago?), it made it much easier to write and compile a correct sequence of

```ts
/** @template T @typedef ... */
/** @templete T @param ... */
function f...
```

And it's easier to explain. The only reason that multiple jsdocs are allowed for `@overload` is that the end of the last overload signature is ambiguous with the beginning of the implementation signature:

```ts
/**
 * @overload   
 * @param {number} x
 * @overload 
 * @param {string} x
 * -beginning of implementation-
 * @param {string | number} x
 */
function f (x) ...
```